server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportGetAllocationStatsAction.java

```diff
@@ -110,23 +113,35 @@ public ActionRequestValidationException validate() {
     public static class Response extends ActionResponse {
 
         private final Map<String, NodeAllocationStats> nodeAllocationStats;
+        private DiskThresholdSettings diskThresholdSettings = null;
```

Comment:
I'd rather this was `final` and assigned on every path through the constructors, and also annotated as `@Nullable`:

```suggestion
        @Nullable // for bwc
        private final DiskThresholdSettings diskThresholdSettings;
```

Reply:
This one is still outstanding